dattri/benchmark/load.py

```diff
@@ -0,0 +1,268 @@
+
+    Args:
+        model (str): The model name for the benchmark setting.
+        dataset (str): The dataset name for the benchmark setting.
+        metric (str): The matrics name for the benchmark setting, which would
+            affected the ground truth. Currently only "lds" and "loo" are
+            supported.
+        download_path (str): The path to download the benchmark files.
+        redownload (bool): Whether to redownload the benchmark files.
+
+    Returns:
+        Tuple[Dict[str, Any], Tuple[torch.Tensor, torch.Tensor]]:
+
+            The first dictionary contains the attribution inputs,
+            the items are listed as following.
+            - "model": The model instance for the benchmark setting.
+            - "models_full": The pre-trained model checkpoints' path with
+                full train dataset, presented as a list of path(str). The
+                models are trained with same hyperparameters and dataset while
+                the only difference is the seed for random initialization.
+            - "models_half": The pre-trained model checkpoints' path with
+                half train dataset, presented as a list of path(str). The
+                models are trained with same hyperparameters while
+                the difference is the dataset sampling (half sampling) for
+                each model checkpoint.
+            - "train_dataset": The path to the training dataset with the
+                same order as the groundtruth's indices.
+            - "test_dataset": The path to the testing dataset with the
+                same order as the groundtruth's indices.
+            - "loss_func": The loss function for the model training. Normally
+                speaking, this should be the same as the target function.
+            - "target_func": The target function for the data attribution. Normally
+                speaking, this should be the same as the loss function.
+            - "train_func": The training function for the model. Normally it's
+                not required if the pre-trained model checkpoints is enough
+                for the algorithm you want to benchmark.
+
+            The second tuple contains the ground truth for the benchmark,
+            the items are listed as following. It can be directly sent to the
+            metrics function defined in `dattri.metrics`. Notably, the groundtruth
+            depends on the `metric` parameter user stated.
+
+    Raises:
+        ValueError: If the model or dataset is not supported.
+    """
+    if model not in SUPPORTED_MODELS:
+        error_msg = f"Model {model} is not supported,\
+                      please choose from {SUPPORTED_MODELS}."
+        raise ValueError(error_msg)
+
+    if dataset not in SUPPORTED_DATASETS:
+        error_msg = f"Dataset {dataset} is not supported,\
+                      please choose from {SUPPORTED_DATASETS}."
+        raise ValueError(error_msg)
+
+    identifier = f"{dataset}_{model}"
```

Comment:
Is it possible that both `model` and `dataset` are supported but the combination is not supported?

Reply:
yeah, we only support limited combination of the two

Reply:
will check this, thx guys

Reply:
directly remove the separate check of model and dataset. make a identifier check to handle this. Good catch, thx.